tests/securedrop/upload_packages.pm

```diff
@@ -0,0 +1,59 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use base "installedtest";
+use strict;
+use testapi;
+use networking;
+use Mojo::File qw(path);
+
+sub upload_package_lists {
+    # Upload inventory of package versions (useful for troubleshooting bad versions)
+    # Borrowed from https://github.com/QubesOS/openqa-tests-qubesos/blob/7242736/tests/update2.pm#L121-L138
+
+    my ($self) = @_;
+
+    x11_start_program('xterm');
+    send_key('alt-f10');  # maximize xterm to ease troubleshooting
+
+    curl_via_netvm;
+
+    my $fname = $self->save_and_upload_log('rpm -qa', 'dom0-packages.txt');
+    my $packages = path('ulogs', $fname)->slurp;
+    $packages = join("\n", sort split(/\n/, $packages));
+    my $all_packages = "Dom0:\n" . $packages;
+    my $templates = script_output('qvm-ls --raw-data --fields name,klass');
+    foreach (sort split /\n/, $templates) {
+        next unless /Template/;
+        s/\|.*//;
+        $fname = $self->save_and_upload_log("qvm-run --no-gui -ap $_ 'rpm -qa; dpkg -l; pacman -Q; true'",
+                "template-$_-packages.txt", ('timeout' => 90, 'failok' => 1));
```

Comment:
I'm not an perl expert by any means, but in other places I see a syntax like this:
```suggestion
                "template-$_-packages.txt", timeout => 90, failok => 1);
```
So, without extra quotes or parenthesis.